File: frontend/src/api/models/DeliveryVisitorDTO.ts
import type { Vehicle } from './Vehicle.ts'

/**
 * @description Dados do visitante
 */
export type DeliveryVisitorDTO = {
  /**
   * @description Nome do visitante
   * @type string
   */
  name: string
  /**
   * @description Documento do visitante
   * @type string
   */
  document: string
  /**
   * @description Telefone do visitante
   * @type string
   */
  phone: string
  /**
   * @type object | undefined
   */
  vehicle?: Vehicle
  /**
   * @description Possui antecedentes criminais
   * @type boolean
   */
  hasCriminalBackgroundCheck: boolean
}

File: backend/src/main/java/br/com/visitsafe/dto/visitor/DeliveryVisitorDTO.java
package br.com.visitsafe.dto.visitor;

import br.com.visitsafe.model.Vehicle;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DeliveryVisitorDTO {

    @NotBlank
    @Schema(description = "Nome do visitante", required = true)
    private String name;

    @NotBlank
    @Schema(description = "Documento do visitante", required = true)
    private String document;

    @NotBlank
    @Schema(description = "Telefone do visitante", required = true)
    private String phone;

    @Schema(description = "Veículo do visitante, se houver", required = false)
    private Vehicle vehicle;

    @NotNull
    @Schema(description = "Possui antecedentes criminais", required = true)
    private Boolean hasCriminalBackgroundCheck;

    public DeliveryVisitorDTO(String name, String document, String phone, Boolean hasCriminalBackgroundCheck) {
        this.name = name;
        this.document = document;
        this.phone = phone;
        this.hasCriminalBackgroundCheck = hasCriminalBackgroundCheck;
        this.vehicle = null;
    }    
}
